### PR TITLE
gcp: retain authenticationInfo.serviceAccountKeyName data

### LINKED ISSUE #10438: [GCP Audit] Preserve all data from the authenticationInfo field
Currently some info in the authenticationInfo field is being dropped by the GCP audit pipeline. We should ensure that this data is preserved to enable efficient investigation and auditing. For example, the `serviceAccountKeyName` field is not preserved and we would like to be able to identify which key is being used to make a request.

https://cloud.google.com/iam/docs/audit-logging/examples-service-accounts#auth-service-account-key
```
{
  "logName": "projects/my-project/logs/cloudaudit.googleapis.com%2Factivity",
  "protoPayload": {
    "@type": "type.googleapis.com/google.cloud.audit.AuditLog",
    "authenticationInfo": {
      "principalEmail": "my-service-account@my-project.iam.gserviceaccount.com",
      "serviceAccountKeyName": "//iam.googleapis.com/projects/my-project/serviceAccounts/my-service-account@my-project.iam.gserviceaccount.com/keys/c71e040fb4b71d798ce4baca14e15ab62115aaef"
    },
    "methodName": "google.cloud.redis.v1.CloudRedis.CreateInstance",
    "request": {
      "@type": "type.googleapis.com/google.cloud.redis.v1.CreateInstanceRequest"
    }
  }
}
```

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

See title.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

- Closes #10438

### CODE REVIEW

File: packages/gcp/data_stream/audit/elasticsearch/ingest_pipeline/default.yml [hunk 1]
```diff
@@ -113,19 +113,23 @@ processors:
 ##
 # email address of authenticated user (redacted) or service account
 # principalEmail -> client.user.email
-  - set:
-      field: client.user.email
-      copy_from: json.protoPayload.authenticationInfo.principalEmail
+  - rename:
```
[the remark this answers is not in the document]
Comment (efd6) [reply packages/gcp/data_stream/audit/elasticsearch/ingest_pipeline/default.yml:116]: Thanks, I was being overly agressive with the renames. I'll fix that.

**Review by muthu-mps — APPROVED** [review]
code owner approval!

**Review by kcreddy — APPROVED** [review]

**Comment by efd6** [comment]
Waiting for @SpencerLN

**Review by kcreddy — APPROVED** [review]

File: packages/gcp/docs/README.md [hunk 2]
```diff
@@ -233,6 +233,9 @@ Please refer to the following [document](https://www.elastic.co/guide/en/ecs/cur
 | gcp.audit.authentication_info.authority_selector | The authority selector specified by the requestor, if any. It is not guaranteed  that the principal was allowed to use this authority. | keyword |
 | gcp.audit.authentication_info.principal_email | The email address of the authenticated user making the request. | keyword |
 | gcp.audit.authentication_info.principal_subject | String representation of identity of requesting party. Populated for both first and third party identities. Only present for APIs that support third-party identities. | keyword |
+| gcp.audit.authentication_info.service_account_delegation_info | The service account key that was used to request the OAuth 2.0 access token. This field identifies the service account key by its full resource name. | flattened |
```
Comment (muthu-mps) [inline packages/gcp/docs/README.md:236]: Both service `account key name` and `delegation` has the same description. Can you please update the description?

Also I don't see the build/README file update for this PR. Is there any reason for not having the changes in the build directory?

**Review by muthu-mps — APPROVED** [review]

**Review by SpencerLN — APPROVED** [review]
LGTM.

**Pull request merged**